### DISCUSSION TITLE
Frontend speed improvement

### DISCUSSION THREAD

**Discussion by Dymirt** [nextcloud/recognize#855] [Ideas]
Hello!
I found that's frontend page 'faces' is fetching all photos for all faces, but shows only one smal pic of this person. May be there is an opportunity for a speed optimisation? Fatching only one main image of a person, and fetching the rest inside person page?

**Comment by Dymirt** [comment]
This is log from my terminal. Just to display a single person face I need to wait until 1235 files will be loaded.
[Debug] [DEBUG] photos: [FetchFacesMixin] Fetched 1235 new files:  – {0: "99030", 1: "99159", 2: "99160", …} (photos-main.js, line 2)

**Comment by Dymirt** [comment]
So I checkt, those files fatching is two times, one when loading 'faces' page, and one when its display photos per person. So there is a real opportunity, just removing fatching all files from 'faces' main page, and fatch only 1 pic per face.

**Reply by marcelklehr** [reply]
Great catch! Would you be able to tackle this?

**Reply by Dymirt** [reply]
Thank you, also I just found this is bug in 'Photos' app, not a 'Recognise'.  the problem seem to be in photos/src/mixins/FetchFacesMixin.js
